liberum_core/src/node/mod.rs

```diff
@@ -116,33 +124,36 @@ impl Node {
     /// the ID of the file using which it can be found.
     #[message]
     pub async fn provide_file(&mut self, path: PathBuf) -> Result<String> {
-        let id = liberum_core::get_file_id(&path)
-            .await
-            .map_err(|e| error!(err = e.to_string(), "Failed to hash file"))
-            .unwrap();
-
         let (resp_send, resp_recv) = oneshot::channel();
 
+        let object: proto::TypedObject = proto::PlainFileObject {
+            name: path.file_name().unwrap().to_str().unwrap().to_string(),
+            content: tokio::fs::read(path).await?,
+        }
+        .into();
+
+        let obj_id = proto::Hash::try_from(&object).unwrap();
+
         self.swarm_sender
             .as_mut()
             .unwrap()
-            .send(SwarmRunnerMessage::ProvideFile {
-                id: id.clone(),
-                path,
+            .send(SwarmRunnerMessage::ProvideObject {
+                object,
+                id: obj_id.clone(),
                 response_sender: resp_send,
             })
             .await?;
 
         resp_recv.await??;
-        let id_str = liberum_core::file_id_to_str(id);
+        let id_str = liberum_core::file_id_hash_to_str(&obj_id.bytes);
 
         Ok(id_str)
     }
 
     #[message]
-    pub async fn download_file(&mut self, id: String) -> Result<Vec<u8>> {
+    pub async fn download_file(&mut self, id: String) -> Result<proto::PlainFileObject> {
         let id_str = id;
```

Comment:
We could get a Hash directly from cli, but the requirement is that it implements From<String>, not TryFrom<String>, and not every string will be valid. As for where to convert the string to Hash using TryFrom, I don't have an opinion on that. It could be done anywhere else